redis/src/commands/json.rs

```diff
@@ -0,0 +1,118 @@
+// can't use rustfmt here because it screws up the file.
+#![cfg_attr(rustfmt, rustfmt_skip)]
+use crate::cmd::{cmd, Cmd, Iter};
+use crate::connection::ConnectionLike;
+use crate::pipeline::Pipeline;
+use crate::types::{FromRedisValue, RedisResult, ToRedisArgs};
+
+#[cfg(feature = "cluster")]
+use crate::commands::ClusterPipeline;
+
+use serde::ser::Serialize;
+
+implement_commands! {
+    'a
+	
+	/// Append the JSON `value` to the array at `path` after the last element in it.
+	fn json_arrappend<K: ToRedisArgs, P: ToString, V: Serialize>(key: K, path: P, value: V) {
```

Comment:
I think `arr_append` would fit with the Rust naming convention better (same for all the other functions).

I don't like the `ToString` argument type here... why can't this just be `&str`? We don't want to require allocations if at all possible.

Also I think this would be a little easier to read if you used `impl Trait` for these arguments (so `key: impl ToRedisArgs` etc instead of the indirection of named generics).

Given that rustfmt is disabled here (I do wonder if that's really still necessary these days), you'll need to spend some effort to manually format this sanely. That means 4-space indents, and the `cmd()` calls should probably be formatted like this:

```rust
        cmd("JSON.ARRAPPEND")
            .arg(key)
            .arg(path.to_string())
            .arg(serde_json::to_string(&value).unwrap())
```

I don't think we want to `unwrap()` here, should just use the `?` operator and add an error variant for `serde_json::Error` that's conditional on the `json` feature.

Reply:
> We don't want to require allocations if at all possible.

Understandable, having a `&str` would do the job better

Reply:
> I don't think we want to `unwrap()` here, should just use the `?` operator and add an error variant for `serde_json::Error` that's conditional on the `json` feature.

Given the current state of the macro, that likely won't be possible and I'll need to make a new macro to do it. I'll get started on that with heavy inspiration from the first macro

Reply:
It would be easier and faster to leave the `unwrap()` in there since the failure only happens if the person writing it doesn't know how json works

https://docs.rs/serde_json/latest/serde_json/fn.to_string.html#errors

Reply:
Progress: I wrote a new macro for json commands. that takes a `RedisResult<Cmd>` in the `$block` as opposed to `&mut Cmd` (needed to add an arg function that takes/returns `self` and returns `self` as opposed to `&mut self`).

All tests pass, and the unwrapping of the results from `$block` is done with `?` before the query which means it still just returns 1 `RedisResult` with 2 chances to fail (serialize error and query error).

I'd recommend reviewing my macro once I open the PR since it's my first time making a macro this complex